sandcastle/api.py

```diff
@@ -481,9 +481,13 @@ def _prepare_mdir_exec(
         unique_dir = root_target_dir / self.pod_name
         script_name = "script.sh"
 
-        # git checkout - oc cp does not preserve the file mode and makes the repo dirty
         # set -e - fail the script when a command fails
-        script_template = f"#!/bin/bash\nset -e\ncd {unique_dir}\nexec {cmd_str}\n"
+        script_template = (
+            "#!/bin/bash\n"
+            "set -e\n"
+            "source /home/sandcastle/setup_env_in_openshift.sh\n"
+            f"cd {unique_dir}\nexec {cmd_str}\n"
+        )
```

Comment:
Why was this change made? Please take a note of this in the commit message.

Reply:
will add a test case for this and further explanation

Reply:
I dropped that comment into a new PR since it started to be pretty big